aws/resource_aws_acm_certificate_test.go

```diff
@@ -0,0 +1,324 @@
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAwsAcmResource_certificateIssuingFlow(t *testing.T) {
+	var conf acm.DescribeCertificateOutput
+	var tags acm.ListTagsForCertificateOutput
+	var confAfterValidation acm.DescribeCertificateOutput
+	if os.Getenv("ACM_CERTIFICATE_ROOT_DOMAIN") == "" {
+		t.Skip("Environment variable ACM_CERTIFICATE_ROOT_DOMAIN is not set")
+	}
+
+	root_zone_domain := os.Getenv("ACM_CERTIFICATE_ROOT_DOMAIN")
+
+	rInt1 := acctest.RandInt()
+
+	domain := fmt.Sprintf("tf-acc-%d.%s", rInt1, root_zone_domain)
+	sanDomain := fmt.Sprintf("tf-acc-%d-san.%s", rInt1, root_zone_domain)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAcmCertificateDestroy,
+		Steps: []resource.TestStep{
+			// Test that we can request a certificate
+			resource.TestStep{
+				Config: testAccAcmCertificateConfig(domain, sanDomain),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAcmCertificateExists("aws_acm_certificate.cert", &conf, &tags),
+					testAccCheckAcmCertificateAttributes("aws_acm_certificate.cert", &conf, domain, sanDomain, "PENDING_VALIDATION"),
+
+					testAccCheckTagsACM(&tags.Tags, "Hello", "World"),
+					testAccCheckTagsACM(&tags.Tags, "Foo", "Bar"),
+
+					resource.TestMatchResourceAttr("aws_acm_certificate.cert", "arn", regexp.MustCompile(`^arn:aws:acm:[^:]+:[^:]+:certificate/.+$`)),
+					resource.TestCheckResourceAttr("aws_acm_certificate.cert", "domain_name", domain),
+					resource.TestCheckResourceAttr("aws_acm_certificate.cert", "subject_alternative_names.#", "1"),
+					resource.TestCheckResourceAttr("aws_acm_certificate.cert", "subject_alternative_names.0", sanDomain),
+					resource.TestCheckResourceAttr("aws_acm_certificate.cert", "tags.%", "2"),
+					resource.TestCheckResourceAttr("aws_acm_certificate.cert", "tags.Hello", "World"),
+					resource.TestCheckResourceAttr("aws_acm_certificate.cert", "tags.Foo", "Bar"),
+				),
+			},
+			// Test that we can change the tags
+			resource.TestStep{
+				Config: testAccAcmCertificateConfigWithChangedTags(domain, sanDomain),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAcmCertificateExists("aws_acm_certificate.cert", &conf, &tags),
+					testAccCheckAcmCertificateAttributes("aws_acm_certificate.cert", &conf, domain, sanDomain, "PENDING_VALIDATION"),
+
+					testAccCheckTagsACM(&tags.Tags, "Environment", "Test"),
+					testAccCheckTagsACM(&tags.Tags, "Foo", "Baz"),
+
+					resource.TestCheckResourceAttr("aws_acm_certificate.cert", "tags.%", "2"),
+					resource.TestCheckResourceAttr("aws_acm_certificate.cert", "tags.Environment", "Test"),
+					resource.TestCheckResourceAttr("aws_acm_certificate.cert", "tags.Foo", "Baz"),
+				),
+			},
+			// Test that validation times out if certificate can't be validated
```

Comment:
Please separate any of the validation testing into `resource_aws_acm_certificate_validation_test.go`

Reply:
Done.